chat-plugins/mafia.js

```diff
@@ -47,59 +48,97 @@
 
 const FS = require('./../lib/fs');
 const LOGS_FILE = 'config/chat-plugins/mafia-logs.json';
+const BANS_FILE = 'config/chat-plugins/mafia-bans.json';
 const MafiaData = require('./mafia-data.js');
 /** @type {MafiaLog} */
 let logs = {leaderboard: {}, mvps: {}, hosts: {}, plays: {}};
+/** @type {MafiaHostBans} */
+let hostBans = Object.create(null);
 /** @type {string[]} */
 let hostQueue = [];
 
 const IDEA_TIMER = 120 * 1000;
 
-try {
-	const json = FS(LOGS_FILE).readIfExistsSync();
-	if (!json) {
-		// Create file
-		FS(LOGS_FILE).writeSync('{"leaderboard": {}, "mvps": {}, "hosts": {}, "plays": {}}');
-	} else {
-		logs = JSON.parse(json);
+/**
+ * @param {string} name
+ */
+function readFile(name) {
+	try {
+		const json = FS(name).readIfExistsSync();
+		if (!json) {
+			writeFile(name, "{}");
+			return false;
+		}
+		return Object.assign(Object.create(null), JSON.parse(json));
```

Comment:
If we bump the minimum Node version, we could start using `{_proto_: null, ...JSON.parse(json)}`

Reply:
How far would we have to bump it? And would it be worth the hassle of raising PS's min node version?

Reply:
I think 8.2 is where object-rest-spread was introduced. I was mostly shitposting there, though. I don't have a strong opinion on which syntax is better, although I do in general prefer doing things syntactically to calling functions; I think it makes things more readable.